src/Amadevus.RecordGenerator.Generators/EqualityUsageAnalyzer.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Amadevus.RecordGenerator.Analyzers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Amadevus.RecordGenerator.Generators
+{
+    internal sealed class EqualityUsageAnalyzer
+    {
+        public static IEnumerable<Diagnostic> GenerateDiagnostics(RecordDescriptor descriptor)
+        {
+            var sealedToken = SyntaxFactory.Token(SyntaxKind.SealedKeyword);
+            if (descriptor.TypeDeclaration.Modifiers.Contains(sealedToken)) yield break;
+
+            yield return Diagnostic.Create(
+                Descriptors.X1001_RecordMustBeSealedIfEqualityIsEnabled,
+                descriptor.TypeDeclarationWithTrivia.Identifier.GetLocation(),
```

Comment:
Why is *WithTrivia necessary?

Reply:
Because I needed to call `GetLocation` on the original identifier, otherwise the warning would appear somewhere in the using declarations.
If I created a property `Location` and called `GetLocation` while creating the Descriptor, there is a chance that the Location isn't even needed and would therefore be calculated unnecessarily.
But it's your decision how we're going to implement it.

Reply:
Right. That makes sense.

With that in mind, please rename those things to something more self-descriptive. `SourceTypeDeclaration` or `OriginalTypeDeclaration`, something like that.